include/nil/crypto3/zk/snark/systems/plonk/placeholder/lookup_argument.hpp

```diff
@@ -51,291 +51,448 @@ namespace nil {
-                                            input_assignment = plonk_columns.constant(lookup.get_vars()[0].index);
-                                            break;
-                                        case VariableType::column_type::selector:
-                                            break;
-                                    }
-                                    switch (lookup_gates[i].constraints[j].lookup_value[k].type) {
-                                        case VariableType::column_type::witness:
-                                            value_assignment = plonk_columns.witness(
-                                                lookup_gates[i].constraints[j].lookup_value[k].index);
-                                            break;
-                                        case VariableType::column_type::public_input:
-                                            value_assignment = plonk_columns.public_input(
-                                                lookup_gates[i].constraints[j].lookup_value[k].index);
-                                            break;
-                                        case VariableType::column_type::constant:
-                                            value_assignment = plonk_columns.constant(
-                                                lookup_gates[i].constraints[j].lookup_value[k].index);
-                                            break;
-                                        case VariableType::column_type::selector:
-                                            break;
-                                    }
-                                    for (std::size_t t = 0; t < basic_domain->m; t++) {
-                                        F_compr_input[t] =
-                                            F_compr_input[t] +
-                                            theta_acc *
-                                                input_assignment[(j + lookup.get_vars()[0].rotation) %
-                                                                 input_assignment.size()] *
-                                                lookup.get_coeff() *
-                                                plonk_columns.selector(lookup_gates[i].selector_index)[t];
-                                        F_compr_value[t] =
-                                            F_compr_value[t] +
-                                            theta_acc * value_assignment[j] *
-                                                plonk_columns.selector(lookup_gates[i].selector_index)[t];
+                        std::vector<math::polynomial_dfs<typename FieldType::value_type>> sorted(reduced_input.size() + reduced_value.size());
+                        for( std::size_t i = 0; i < sorted.size(); i++ ){
+                            sorted[i] = math::polynomial_dfs<typename FieldType::value_type>(basic_domain->m-1, basic_domain->m, FieldType::value_type::zero());
```

Comment:
You assign the same value to this vector. It's probably faster to say std::vector<math::polynomial_dfs<typename FieldType::value_type>> sorted(reduced_input.size() + reduced_value.size(), 

math::polynomial_dfs<typename FieldType::value_type>(basic_domain->m-1, basic_domain->m, FieldType::value_type::zero()));